tools/Google.Cloud.Tools.VersionCompat/Detectors/ClassAndInterface.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Tools.VersionCompat.CecilUtils;
+using Google.Cloud.Tools.VersionCompat.Utils;
+using Mono.Cecil;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Google.Cloud.Tools.VersionCompat.Detectors
+{
+    internal class ClassAndInterface
+    {
+        public ClassAndInterface(TypeDefinition o, TypeDefinition n) => (_o, _n) = (o, n);
+
+        private readonly TypeDefinition _o;
+        private readonly TypeDefinition _n;
+
+        public IEnumerable<Diff> ImplementedInterfaces()
+        {
+            // Treat any change to exported implemented interfaces is breaking.
+            // This isn't correct, as adding or removing an interface which an another interface itself implements is OK.
```

Comment:
And I'd say that adding an implemented interface in a concrete class is okay anyway... it would be okay for a type to add an implementation of `IEquatable<T>` for example.

Yes, in the strictest sense that could be breaking by changing overload resolution - but I suspect that given *any* public API change I could find some way it's breaking.

It may be that we should increase the separation between "Here are the semantic differences" and "This is what I consider to be breaking and what isn't" pretty soon, as I suspect the more code we write, the more there will be to change...

Reply:
Yes, I defined this as major breaking due to the overload ambiguity issue you mention.

However, I agree we need to define a separation between "theoretically breaking" and "pragmatically this is OK".

I think I'll leave this as it is until we have a better-defined process.